pkg/cmd/experimental/syncgroups/cli/prunegroups.go

```diff
@@ -0,0 +1,276 @@
+package cli
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+
+	"github.com/spf13/cobra"
+
+	configapilatest "github.com/openshift/origin/pkg/cmd/server/api/latest"
+	kapi "k8s.io/kubernetes/pkg/api"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	kerrs "k8s.io/kubernetes/pkg/util/errors"
+	kyaml "k8s.io/kubernetes/pkg/util/yaml"
+
+	"github.com/openshift/origin/pkg/auth/ldaputil"
+	osclient "github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/experimental/syncgroups"
+	"github.com/openshift/origin/pkg/cmd/experimental/syncgroups/interfaces"
+	"github.com/openshift/origin/pkg/cmd/server/api"
+	"github.com/openshift/origin/pkg/cmd/server/api/validation"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+)
+
+const (
+	PruneGroupsRecommendedName = "prune-groups"
+
+	pruneGroupsLong = `
+Prune OpenShift Groups referencing records from an external provider.
+
+In order to prune OpenShift Group records using those from an external provider, determine which Groups you wish
+to prune. For instance, all or some groups may be selected from the current Groups stored in OpenShift that have
+been synced previously. Any combination of a literal whitelist, a whitelist file and a blacklist file is supported.
+The path to a sync configuration file that was used for syncing the groups in question is required in order to 
+describe how data is requested from the external record store. Default behavior is to prune all OpenShift groups 
+for which the LDAP record does not exist.
+`
+	pruneGroupsExamples = `  # Prune all orphaned groups 
+  $ %[1]s --sync-config=/path/to/ldap-sync-config.yaml
+
+  # Prune all orphaned groups except the ones from the blacklist file
+  $ %[1]s --blacklist=/path/to/blacklist.txt --sync-config=/path/to/ldap-sync-config.yaml
+
+  # Prune all orphaned groups from a list of specific groups specified in a whitelist file
+  $ %[1]s --whitelist=/path/to/whitelist.txt --sync-config=/path/to/ldap-sync-config.yaml
+
+  # Prune all orphaned groups from a list of specific groups specified in a whitelist
+  $ %[1]s groups/group_name groups/other_name --sync-config=/path/to/ldap-sync-config.yaml
+`
+)
+
+type PruneGroupsOptions struct {
+	// Config is the LDAP sync config read from file
+	Config api.LDAPSyncConfig
```

Comment:
Chosen for compatibility, the entire sync config is not necessary for the prune job.